cluster-autoscaler/provisioningrequest/provreqclient/client.go

```diff
@@ -178,30 +178,29 @@ func newPodTemplatesLister(client *kubernetes.Clientset, stopChannel <-chan stru
 	return podTemplLister, nil
 }
 
-// ProvisioningRequestForPods check that all pods belong to one ProvisioningRequest and return it.
-func ProvisioningRequestForPods(client *ProvisioningRequestClient, unschedulablePods []*apiv1.Pod) (*provreqwrapper.ProvisioningRequest, error) {
+// ProvisioningRequestsForPods check that all pods belong to one ProvisioningRequest and return it.
+func ProvisioningRequestsForPods(client *ProvisioningRequestClient, unschedulablePods []*apiv1.Pod) ([]*provreqwrapper.ProvisioningRequest, error) {
+	prMap := make(map[string]*provreqwrapper.ProvisioningRequest)
+	prList := []*provreqwrapper.ProvisioningRequest{}
 	if len(unschedulablePods) == 0 {
-		return nil, fmt.Errorf("empty unschedulablePods list")
-	}
-	if unschedulablePods[0].OwnerReferences == nil || len(unschedulablePods[0].OwnerReferences) == 0 {
-		return nil, fmt.Errorf("pod %s has no OwnerReference", unschedulablePods[0].Name)
-	}
-	provReq, err := client.ProvisioningRequest(unschedulablePods[0].Namespace, unschedulablePods[0].OwnerReferences[0].Name)
-	if err != nil {
-		return nil, fmt.Errorf("failed retrive ProvisioningRequest from unscheduled pods, err: %v", err)
+		return prList, nil
 	}
 	for _, pod := range unschedulablePods {
-		if pod.Namespace != unschedulablePods[0].Namespace {
-			return nil, fmt.Errorf("pods %s and %s are from different namespaces", pod.Name, unschedulablePods[0].Name)
-		}
 		if pod.OwnerReferences == nil || len(pod.OwnerReferences) == 0 {
 			return nil, fmt.Errorf("pod %s has no OwnerReference", pod.Name)
```

Comment:
Perhaps rather than failing here we can just ignore those pods and log an error?